requirements.txt

```diff
@@ -1,5 +1,5 @@
 # pinned dependencies to reproduce an entire development environment to use HDMF-ZARR
-hdmf==3.5.4
+hdmf==3.7.0
```

Comment:
Hey guys, is there a reason why you use exact version pinning here? It's causing problems in https://github.com/NeurodataWithoutBorders/nwbinspector/pull/388 and could cause problems elsewhere as well

Would there be catastrophic issues caused by looser versioning? e.g., `hdmf>=3.5.4`

Reply:
@rly @oruebel @mavaylon1

Reply:
It's to allow HDMF_ZARR to use the newest external resources, but I really don't have time this week to fix anything backend breaking until this week, which is too close to the dev/user days. I believe @oruebel was going to look into it last week.

Reply:
I will start at the end of the week but I will make this a project for the dev days.